sub/sd_ass.c

```diff
@@ -259,8 +261,12 @@ static void decode(struct sd *sd, struct demux_packet *packet)
             packet->duration = UNKNOWN_DURATION;
         }
         char **r = lavc_conv_decode(ctx->converter, packet);
-        for (int n = 0; r && r[n]; n++)
-            ass_process_data(track, r[n], strlen(r[n]));
+        for (int n = 0; r && r[n]; n++) {
+            if (sd->opts->sub_filter_SDH)
+                filter_SDH(sd, track->event_format, 0, r[n] , sd->opts->sub_filter_SDH_harder);
```

Comment:
Maybe not, but it avoids allocating memory just for a possibly rewritten subtitle line.
One way could be to have the filter only read original line and allocate a new and write to it.
Combining that with the call for ASS subtitles to the filter where you may prefer that I do not use a reusable buffer, there would be one more allocate/copy that is immediately discarded.
To avoid that I would have to change the filter to instead of having null terminated input string use length to tell when subtitle ends.